src/tools/diagnostics.md

```diff
@@ -0,0 +1,2436 @@
+---
+title: Diagnostics
+description: Details for diagnostics produced by the Dart analyzer.
+---
+
+This page lists all diagnostic messages produced by the Dart analyzer,
+with details about what those messages mean
+and how you can fix your code.
+For more information about the analyzer, see
+[Customizing static analysis](/guides/language/analysis-options).
+
+
+## Glossary
+
+This page uses the following terms.
+
+### Element
+
+[PENDING: definition]
+
+### Potentially non-nullable
+
+A type is _potentially non-nullable_ if it's either explicitly non-nullable or
+if it's a type parameter.
+The latter case is included because
+the actual runtime type might be non-nullable.
+
+
+## Compile-time errors
+
+The analyzer produces the following diagnostics when [PENDING...].
+
+### AMBIGUOUS_SET_OR_MAP_LITERAL_BOTH
+
+#### Message
+
+This literal contains both 'Map' and 'Iterable' spreads, which makes it impossible to determine whether the literal is a map or a set.
+
+#### Description
+
+Because map and set literals use the same delimiters (`{` and `}`), the analyzer looks at the type arguments and the elements to determine which kind of literal you meant. When there are no type arguments and all of the elements are spread elements (which are allowed in both kinds of literals), then the analyzer uses the types of the expressions that are being spread. If all of the expressions have the type `Iterable`, then it's a set literal; if they all have the type `Map`, then it's a map literal.
```

Comment:
Our styles aren't changing the background color for monospaced codeblocks, so what should look acceptable here e.g. :

the same delimiters (`{` and `}`), the analyzer...

ends up looking like this:
<img width="68" alt="Screen Shot 2019-05-31 at 3 29 17 PM" src="https://user-images.githubusercontent.com/5550175/58738089-2be30f00-83b9-11e9-84de-de2e4f27b6c7.png">

(The above is a little jarring to my eye, but disregard this comment if it's palatable for everyone else)

One workaround is to find a way to craft the sentence without using the parens, e.g. :

Because map and set literals use the same delimiters `{` and `}` the analyzer looks...

Reply:
It looks odd to me to drop the parens, but I'm not a professional writer, so I'll go with whatever the two of you decide.